core/src/main/java/org/jboss/pnc/build/finder/core/BuildFinder.java

```diff
@@ -595,11 +596,17 @@ private KojiBuild findBestBuildFromCandidates(List<KojiBuild> candidates, List<K
                     buildCache.put(cacheRpmBuildInfo.getBuildInfo().getId(), cacheRpmBuildInfo);
                 }
             } else {
-                if (cacheManager == null || (cacheArchiveInfos = checksumCaches.get(ChecksumType.md5)
-                        .get(checksum.getValue())) == null) {
+                ListKojiArchiveInfoProtobufWrapper wrapper = null;
+
+                if (checksumCaches != null) {
+                    wrapper = checksumCaches.get(ChecksumType.md5).get(checksum.getValue());
```

Comment:
So, it's only adding an empty map for checksums that are specified in the configuration as per https://github.com/project-ncl/build-finder/blob/master/core/src/main/java/org/jboss/pnc/build/finder/core/DistributionAnalyzer.java#L115.

However,  only `md5` checksums are put on the queue for `BuildFinder`, so I thought `md5` was forced to exist somewhere else, since it cannot find builds without `md5` (since Koji requires it, and since it's the chosen one for PNC even though PNC technically supports others).

Right now I am not sure if we can even get a `NullPointerException`, but might as well use the same pattern as the rest of the code.